generator/intersection.go

```diff
@@ -37,24 +37,82 @@ func CreateCatalogsFromProfile(profile *profile.Profile) []*catalog.Catalog {
 			logrus.Errorf("cannot parse catalog listed in import.href %v", err)
 		}
 		//Prepare a new catalog object to merge into the final List of OutputCatalogs
-		newCatalog := getMappedCatalogControlsFromImport(importedCatalog, profileImport)
+		newCatalog, err := getMappedCatalogControlsFromImport(importedCatalog, profileImport)
+		if err != nil {
+			return nil, err
+		}
 		outputCatalogs = append(outputCatalogs, &newCatalog)
 	}
-	return outputCatalogs
+	return outputCatalogs, nil
 }
 
-func mapSubControls(mappedCatalogs *catalog.Catalog, calls []profile.Call) {
-	for i, g := range mappedCatalogs.Groups {
-		for j, control := range g.Controls {
-			for _, call := range calls {
-				if strings.ToLower(control.Id) == strings.ToLower(getControlIDFromSubControl(call.SubcontrolId)) {
-					mappedCatalogs.Groups[i].Controls[j].Subcontrols = append(mappedCatalogs.Groups[i].Controls[j].Subcontrols, catalog.Subcontrol{
-						Id: call.SubcontrolId,
-					})
+func getGroupID(g *catalog.Group) (string, error) {
```

Comment:
As discussed, this logic is NIST-800-53 specific, it should be generalized. Let's discuss this in person.